python/lsst/sconsUtils/scripts.py

```diff
@@ -102,15 +109,28 @@ def initialize(cls, packageName, versionString=None, eupsProduct=None, eupsProdu
             except TypeError:
                 pass
             state.targets["version"] = state.env.VersionModule(versionModuleName)
+        scripts = []
         for root, dirs, files in os.walk("."):
             if "SConstruct" in files and root != ".":
                 dirs[:] = []
                 continue
             dirs[:] = [d for d in dirs if not d.startswith('.')]
             dirs.sort()  # happy coincidence that include < libs < python < tests
             if "SConscript" in files:
-                state.log.info("Using Sconscript at %s/SConscript" % root)
-                SConscript(os.path.join(root, "SConscript"))
+                scripts.append(os.path.join(root, "SConscript"))
+        def cmpOrder(a, b):
+            ai = len(sconscriptOrder)
+            bi = len(sconscriptOrder)
+            for i, item in enumerate(sconscriptOrder):
+                if a.startswith(item):
+                    ai = i
+                if b.startswith(item):
+                    bi = i
+            return cmp(ai, bi)
+        scripts.sort(cmp=cmpOrder)
```

Comment:
Does `key` take a function in Python 3?  This implementation won't work at all unless that's the case.